### PR TITLE
Changes for RancherOSv2

### PR DESCRIPTION
**Pull request by bk201**
**Problem:**
We need to support RancherOSv2.

**Solution:**
The PR contains changes needed for [RancherOSv2 integration](https://github.com/harvester/harvester-installer/pull/93).
- Remove NetworkAttachmentDefinition CRD registration.
- Update the node promotion commands.

**Related Issue:**
https://github.com/harvester/harvester/issues/581

**Test plan:**
- Regression tests
- Node promotion: Make sure when there are more than 3 nodes, 3 nodes are provisioned as server role. Example:
  ```
  node2:~ # kubectl get nodes
  NAME    STATUS   ROLES                       AGE   VERSION
  node1   Ready    control-plane,etcd,master   55m   v1.21.2+rke2r1
  node2   Ready    control-plane,etcd,master   46m   v1.21.2+rke2r1
  node3   Ready    control-plane,etcd,master   37m   v1.21.2+rke2r1
  ```

### CODE REVIEW

File: deploy/charts/harvester/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: harvester-helpers
+  labels:
+{{ include "harvester.labels" . | indent 4 }}
+data:
+  promote.sh: |
+    #!/bin/bash -ex
+    echo start promote
```
Comment (gitlawr) [inline deploy/charts/harvester/templates/configmap.yaml]: Is it simpler to package the scripts in the harvester server image? Or we want it to be configurable somehow

**Review by gitlawr — APPROVED** [review]
LGTM

**Comment by bk201** [comment]
Rebased.

@gitlawr Please take a look again. Sometimes a promoting node doesn't have metadata (especially all nodes are started at the same time), I [add some code ](https://github.com/harvester/harvester/pull/1033/files#diff-576472fe7f6767e351ace8587796e52b928c66be6eeb5ab7fd32cb9b8127f061R116-R120)to detect it. Thanks!

File: pkg/controller/master/node/promote_controller.go [hunk 2]
```diff
@@ -115,6 +113,11 @@ func (h *PromoteHandler) OnNodeChanged(key string, node *corev1.Node) (*corev1.N
 		return node, nil
 	}
 
+	if promoteNode.Kind == "" || promoteNode.APIVersion == "" {
```
Comment (gitlawr) [inline pkg/controller/master/node/promote_controller.go:119]: Can you add a comment for the check

Comment (bk201) [reply pkg/controller/master/node/promote_controller.go:119]: added. Thanks.

**Review by gitlawr — APPROVED** [review]
LGTM

**Comment by guangbochen** [comment]
since the `multus-cni` is installed by default of the rancherd/rke2 integration, so we should remove with the new OS change.

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**